File: src/stack/ImplementQueueUsingStacks.java
package stack;
/*
Implement the following operations of a queue using stacks.

push(x) -- Push element x to the back of queue.
pop() -- Removes the element from in front of queue.
peek() -- Get the front element.
empty() -- Return whether the queue is empty.
Notes:
You must use only standard operations of a stack -- which means only push to top, peek/pop from top, size, and is empty operations are valid.
Depending on your language, stack may not be supported natively. You may simulate a stack by using a list or deque (double-ended queue), as long as you use only standard operations of a stack.
You may assume that all operations are valid (for example, no pop or peek operations will be called on an empty queue).
 * */

import java.util.Stack;

public class ImplementQueueUsingStacks 
{
	Stack<Integer> queue = new Stack<>();
	
    // Push element x to the back of queue.
    public void push( int x ) 
    {
        Stack<Integer> temp = new Stack<>();
		while ( !queue.empty() )
		{
			temp.push( queue.pop() );
		}
		queue.push( x );
		while ( !temp.empty() )
		{
			queue.push( temp.pop() );
		}
    }

	// Removes the element from in front of queue.
	public void pop()
	{
		queue.pop();
	}

	// Get the front element.
	public int peek()
	{
		return queue.peek();
	}

	// Return whether the queue is empty.
	public boolean empty()
	{
		return queue.empty();
	}
}

File: Python/Stack/ImplementQueueUsingStacks.py
# Definition for a binary tree node.
import heapq
import unittest

# Read about enumerate in python
from collections import defaultdict
from typing import List


class MyQueue:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.inStack = []
        self.outStack = []

    def push(self, x: int) -> None:
        """
        Push element x to the back of queue.
        """
        self.inStack.append(x)

    def pop(self) -> int:
        """
        Removes the element from in front of queue and returns that element.
        """
        result = self.peek()
        self.outStack.pop()
        return result

    def peek(self) -> int:
        """
        Get the front element.
        """
        if not self.outStack:
            while self.inStack:
                self.outStack.append(self.inStack.pop())
        return self.outStack[-1]

    def empty(self) -> bool:
        """
        Returns whether the queue is empty.
        """
        return not self.inStack and not self.outStack

class ImplementQueueUsingStacks(unittest.TestCase):

    def test_Leetcode(self):
        return

if __name__ == '__main__':
    unittest.main()